embassy-stm32/src/usart/v3.rs

```diff
@@ -99,7 +115,11 @@ impl<'d, T: Instance> Uart<'d, T> {
     }
 }
 
-impl<'d, T: Instance> embedded_hal::blocking::serial::Write<u8> for Uart<'d, T> {
+impl<'d, T: Instance, TxDma, RxDma> embedded_hal::blocking::serial::Write<u8>
+    for Uart<'d, T, TxDma, RxDma>
```

Comment:
You can directly specify the NoDma type to restrict the impl. This way you don't need the `NoDmaMarker` trait.

`impl<'d, T: Instance,  RxDma> embedded_hal::blocking::serial::Write<u8> for Uart<'d, T, NoDma, RxDma>`